sklearn/metrics/_ranking.py

```diff
@@ -668,6 +668,12 @@ def precision_recall_curve(y_true, probas_pred, pos_label=None,
     array([0.35, 0.4 , 0.8 ])
 
     """
+    # ignore weights equal zero
+    indexes_zeros = sample_weight.index(0)
+    del y_true[indexes_zeros]
+    del probas_pred[indexes_zeros]
+    del sample_weight[indexes_zeros]
```

Comment:
All those arguments are more expected to be numpy arrays instead of lists. So instead we should write numpy code such as follows:

```
if sample_weight is not None:
    if not np.all(sample_weight >= 0)
        raise ValueError("negative values in sample_weight are invalid")

    nonzero_weight_mask = sample_weight > np.finfo(np.float32).eps
    y_true = y_true[nonzero_mask]
    probas_pred = probas_pred[nonzero_mask]
    sample_weight = sample_weight[nonzero_mask]
```

But this should only be done after we call `check_array`, `column_or_1d` or similar validation functions on all the arguments. Those checks are centralized in the `_binary_clf_curve` function itself, so the zero weight filtering code should probably be moved to the body of that function.

Reply:
Thank you very much, Olivier. I have added a check_negative at _check_sample_weight which partially answers #15531. I have added some lines to check that it's not zero.